inference-engine/src/mkldnn_plugin/emitters/jit_snippets_emitters.hpp

```diff
@@ -26,39 +46,56 @@ class KernelEmitter : public jit_emitter {
                    const std::vector<size_t>& pool,
                    const std::vector<size_t>& gpr,
                    const MKLDNNPlugin::emitter_context *emit_context) const override {
-        auto tile_rank = in[0]; // count of tile dimensions
-        auto nparams = in[1];
+        const size_t tile_rank = in[0]; // count of tile dimensions
+        const size_t num_inputs = in[1];
+        const size_t num_outputs = in[2];
+        const size_t num_params = num_inputs + num_outputs;
         int reg64_tmp_start { 8 }; // R8, R9, R10, R11, R12, R13, R14, R15 inputs+outputs+1
-        Xbyak::Reg64 params   { dnnl::impl::cpu::x64::abi_param1 };
-        Xbyak::Reg64 schedule { dnnl::impl::cpu::x64::abi_param2 };
-        Xbyak::Reg64 offsets_ { dnnl::impl::cpu::x64::abi_param3 };
-        Xbyak::Reg64 offsets  { dnnl::impl::cpu::x64::abi_not_param1 };
+        Xbyak::Reg64 reg_indexes   { dnnl::impl::cpu::x64::abi_param1 };
+        Xbyak::Reg64 reg_const_params { dnnl::impl::cpu::x64::abi_param2 };
+        #ifdef _WIN32
+        Xbyak::Reg64 reg_tmp_64 { Xbyak::Operand::RDI };
+        #else
+        Xbyak::Reg64 reg_tmp_64 { Xbyak::Operand::RCX };
+        #endif
 
         if (tile_rank != 2)
             IE_THROW() << "Kernel of codegen supports only Tile2D" << std::endl;
 
         h->preamble();
 
-        // to avoid register conflicts on Win systems
-        h->mov(offsets, offsets_);
+        std::vector<Xbyak::Reg64> regs(num_params);
 
-        // ptrs
-        std::vector<Xbyak::Reg64> regs(nparams);
-        for (auto i = 0; i < nparams; i++) {
+        const int64_t harness_num_dims = _jep.output_dims.size() - 1;
+        auto init_ptrs_with_offsets = [&](Reg64 pointer, const int64_t *offsets) {
+            for (int j = 0; j < harness_num_dims; j++) {
+                if (_jep.output_dims[j] != 1 && offsets[j] != 0) {
+                    h->mov(reg_tmp_64, offsets[j]);
+                    h->imul(reg_tmp_64, h->ptr[reg_indexes + j * sizeof(size_t)]);
+                    h->add(pointer, reg_tmp_64);
+                }
+            }
+        };
+        for (auto i = 0; i < num_params; i++) {
             regs[i] = Xbyak::Reg64(reg64_tmp_start + i);
-            h->mov(regs[i], h->ptr[params + i * sizeof(int64_t)]);
+            if (i < num_inputs)
+                h->mov(regs[i], h->ptr[reg_const_params + GET_OFF(src_ptrs) + i * sizeof(void*)]);
+            else
+                h->mov(regs[i], h->ptr[reg_const_params + GET_OFF(dst_ptrs) + (i - num_inputs) * sizeof(void*)]);
+            init_ptrs_with_offsets(regs[i], &_jep.data_offsets[i * harness_num_dims]);
         }
 
         // external amount
-        Xbyak::Reg64 amount = Xbyak::Reg64(reg64_tmp_start + nparams);
-        h->mov(amount, h->ptr[schedule + (tile_rank - 1) * sizeof(int64_t)]);
+        Xbyak::Reg64 amount = Xbyak::Reg64(reg64_tmp_start + num_params);
+        h->mov(amount, _jep.scheduler_dims[tile_rank-1]);
```

Comment:
As I remembered, We wanted to reverse scheduler dims..

Reply:
Yes, we did. Fixed, so now inner dimension is the last one.